Merge contracts and scripts from rundler-hc repo

mmontour1306 opened the pull request:
This PR moves the HC contracts, the reference offchain-rpc implementation, and associated Python scripts into this repo. Some functions were added to the hybrid_compute_sdk package itself, while other code is kept external. The version of hybrid_compute_sdk is bumped to 0.2.44.

Further reorganization might be appropriate, e.g. splitting offchain-rpc out into its own package.

The HCHelper.sol and HybridAccount.sol contracts in this PR contain an update to support an auto-registration protocol. The contracts deployed in the field have not yet been updated to this version.

Other instances of the core contracts have been removed, to remove confusion about which version is the official one. However I have not yet updated existing examples which might be looking for the contracts in their old location. I'm therefore setting this PR to 'draft' for now.

mmontour1306 committed:
refactor: move HC contracts from rundler-hc to this repo.

These contracts include support for the auto-registration protocol.

Changes to be committed:
	new file:   packages/contracts/v0_7/.gitignore
	new file:   packages/contracts/v0_7/README.md
	new file:   packages/contracts/v0_7/foundry.toml
	new file:   packages/contracts/v0_7/hc_examples/TestAuctionSystem.sol
	new file:   packages/contracts/v0_7/hc_examples/TestCaptcha.sol
	new file:   packages/contracts/v0_7/hc_examples/TestHybrid.sol
	new file:   packages/contracts/v0_7/hc_examples/TestKyc.sol
	new file:   packages/contracts/v0_7/hc_examples/TestRainfallInsurance.sol
	new file:   packages/contracts/v0_7/hc_examples/TestRandom.sol
	new file:   packages/contracts/v0_7/hc_examples/TestSportsBetting.sol
	new file:   packages/contracts/v0_7/hc_examples/TestTokenPrice.sol
	new file:   packages/contracts/v0_7/hc_src/HCHelper.sol
	new file:   packages/contracts/v0_7/hc_src/HybridAccount.sol
	new file:   packages/contracts/v0_7/hc_src/HybridAccountFactory.sol
	new file:   packages/contracts/v0_7/hc_src/SimplePaymaster.sol
	new file:   packages/contracts/v0_7/lib/chainlink/VRF.sol

mmontour1306 committed:
refactor: remove other versions of HC contracts.

The ones in packages/contracts/v0_7/hc_src/ should be used for
new development. Symlinks could be used as a temporary measure
to support applications expecting the contracts to be at their
old path.

Changes to be committed:
	deleted:    packages/contracts/core/0_7/HCHelper.sol
	deleted:    packages/contracts/core/0_7/HybridAccount.sol
	deleted:    packages/contracts/core/0_7/HybridAccountFactory.sol
	deleted:    packages/contracts/core/HCHelper.sol
	deleted:    packages/contracts/samples/HybridAccount.sol
	deleted:    packages/contracts/samples/HybridAccountFactory.sol

mmontour1306 committed:
refactor: move the offchain_rpc files from rundler-hc repo to here.

The server is now configured to load offchain handlers dynamically from
a directory. Also reorganized the old "userop.py" stuff into "test.py"
and "tests/*.py" (does not have the auto-detection and loading).

Changes to be committed:
	new file:   packages/server-python/offchain_rpc/Dockerfile
	new file:   packages/server-python/offchain_rpc/handlers/README.md
	new file:   packages/server-python/offchain_rpc/handlers/add_sub_2_offchain.py
	new file:   packages/server-python/offchain_rpc/handlers/auction_system_offchain.py
	new file:   packages/server-python/offchain_rpc/handlers/check_kyc_offchain.py
	new file:   packages/server-python/offchain_rpc/handlers/ramble_offchain.py
	new file:   packages/server-python/offchain_rpc/handlers/sports_betting_offchain.py
	new file:   packages/server-python/offchain_rpc/handlers/sys_register_caller.py
	new file:   packages/server-python/offchain_rpc/handlers/vrf_offchain.py
	new file:   packages/server-python/offchain_rpc/offchain.py
	new file:   packages/server-python/offchain_rpc/readme.md
	new file:   packages/server-python/offchain_rpc/readme_tests.md
	new file:   packages/server-python/offchain_rpc/requirements.txt
	new file:   packages/server-python/offchain_rpc/test.py
	new file:   packages/server-python/offchain_rpc/test_utils.py
	new file:   packages/server-python/offchain_rpc/tests/add_sub_2_test.py
	new file:   packages/server-python/offchain_rpc/tests/auction_system_test.py
	new file:   packages/server-python/offchain_rpc/tests/check_kyc_test.py
	new file:   packages/server-python/offchain_rpc/tests/ramble_test.py
	new file:   packages/server-python/offchain_rpc/tests/sports_betting_test.py
	new file:   packages/server-python/offchain_rpc/tests/vrf_test.py

mmontour1306 committed:
refactor: move deployer scripts etc. from rundler-hc repository

A local devnet can now be deployed from this repository. Some code
has been placed into "deploy.py" in the SDK package.

Changes to be committed:
	new file:   .gitignore
	new file:   README.md
	new file:   aa-hc-check.sh
	new file:   aa_client.py
	new file:   deploy_examples.py
	new file:   deploy_local.py
	new file:   docker-compose.yml
	new file:   forge_scripts/ExampleDeploy_v7.s.sol
	new file:   forge_scripts/LocalDeploy_v7.s.sol
	new file:   kms-init/seed.yaml
	new file:   local.env
	new file:   runit.sh

mmontour1306 committed:
refactor: merge some code from rundler-hc into the hybrid_compute_sdk package

This adds functions to support the local devnet Deployer and the
examples/tests from rundler-hc/hybrid_compute (formerly "userop.py")

Changes to be committed:
	modified:   packages/server-python/hybrid_compute_sdk/__init__.py
	new file:   packages/server-python/hybrid_compute_sdk/aa_utils.py
	new file:   packages/server-python/hybrid_compute_sdk/deploy.py
	modified:   packages/server-python/hybrid_compute_sdk/server.py
	modified:   packages/server-python/hybrid_compute_sdk/utils/userop_utils.py
	modified:   packages/server-python/setup.py

mmontour1306 committed:
refactor: add a github action to build the offchain-rpc image

This could be split out into its own repository in the future.

Changes to be committed:
	new file:   .github/workflows/docker-images.yaml

wsdt commented:
- packages/deployer/local.env (just to double check: they are all test/local values?)
- packages/server-python/hybrid_compute_sdk/aa_utils.py (fallback https://sepolia.boba.network is deprecated, I presume)
- There are now 2 versions of aa_utils, hybrid_compute_sdk/utils/aa_utils and hybrid_compute_sdk/aa_utils which one to keep?
- contracts v0_7 structure looks way cleaner now!

Will also test it on my end!

wsdt reviewed (approved):
left a comment

mmontour1306 commented:
The values in local.env are for use on a local devnet. This file was moved from https://github.com/bobanetwork/rundler-hc/blob/boba-develop/hybrid-compute/local.env with some variables rearranged.

I will fix the duplicate "aa_utils" and update the branch.

mmontour1306 committed:
refactor: remove old files from utils

Changes to be committed:
	deleted:    packages/server-python/hybrid_compute_sdk/utils/aa_utils.py
	deleted:    packages/server-python/hybrid_compute_sdk/utils/userop_utils.py

mmontour1306 commented:
Kept "packages/server-python/hybrid_compute_sdk/aa_utils.py" and removed old ones from "utils". The "utils/userop_utils.py" was an old version of what's now "offchain_rpc/test_utils.py".
